### PR TITLE
adding pagination of repositories on bitbucketserver

### PR DESCRIPTION
**Pull request by thiago**
Fix issue #1776

### CODE REVIEW

File: remote/bitbucketserver/internal/client.go [hunk 1]
```diff
@@ -213,3 +196,34 @@ func (c *Client) doDelete(url string) error {
...
+	contents, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	var repoResponse Repos
+	err = json.Unmarshal(contents, &repoResponse)
+	if err != nil {
+		return nil, err
+	}
+	log.Debug(fmt.Printf("repoResponse: %+v\n", repoResponse))
```
Comment (bradrydzewski) [inline remote/bitbucketserver/internal/client.go]: while in the code, some minor things that could be improved:

```
log.Debugf("repoResponse: %+v\n", repoResponse)
```

File: remote/bitbucketserver/internal/client.go [hunk 2]
```diff
@@ -213,3 +196,34 @@ func (c *Client) doDelete(url string) error {
...
+		return nil, err
+	}
+	defer response.Body.Close()
+	contents, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	var repoResponse Repos
+	err = json.Unmarshal(contents, &repoResponse)
+	if err != nil {
```
Comment (bradrydzewski) [inline remote/bitbucketserver/internal/client.go]: while in the code, some minor things that could be improved:

```diff
-err = json.Unmarshal(contents, &repoResponse)
+err = json.NewDecoder(response.Body).Decode(&repoResponse)
```

Comment (thiago) [inline remote/bitbucketserver/internal/client.go]: Sure, thanks for your attention!

**Comment by josmo** [comment]
LGTM :)  Thanks @thiago

**Comment by appleboy** [comment]
LGTM! 👍

**Comment by josmo** [comment]
Btw not sure on you're end but making the limit 400 instead of 1000 may make it a little more responsive. I haven't tried it but might be worth checking out

**Comment by thiago** [comment]
it does not make it slower?

**Comment by bradrydzewski** [comment]
> Btw not sure on you're end but making the limit 400 instead of 1000 may make it a little more responsive. I haven't tried it but might be worth checking out

I'll merge and this can be handled in a follow-up pull request. The current implementation could probably be improved a bit more to more closely mirror the existing bitbucket implementation, but this can be done separately:

```
func (c *Client) ListRepos(account string, opts *ListOpts) (*RepoResp, error) {
	out := new(RepoResp)
	uri := fmt.Sprintf(pathRepos, c.base, account, opts.Encode())
	err := c.do(uri, get, nil, out)
	return out, err
}

func (c *Client) ListReposAll(account string) ([]*Repo, error) {
	var page = 1
	var repos []*Repo

	for {
		resp, err := c.ListRepos(account, &ListOpts{Page: page, PageLen: 100})
		if err != nil {
			return repos, err
		}
		repos = append(repos, resp.Values...)
		if len(resp.Next) == 0 {
			break
		}
		page = resp.Page + 1
	}
	return repos, nil
}
```

**Pull request merged**